addons/binding/org.openhab.binding.kodi/src/main/java/org/openhab/binding/kodi/handler/KodiHandler.java

```diff
@@ -0,0 +1,310 @@
+                if (command instanceof StringType) {
+                    playURI(command);
+                } else if (command.equals(RefreshType.REFRESH)) {
+                    // updateState(CHANNEL_PLAYURI, new StringType(""));
+                }
+                break;
+            case CHANNEL_SHOWNOTIFICATION:
+                if (command instanceof StringType) {
+                    connection.showNotification(command.toString());
+                } else if (command.equals(RefreshType.REFRESH)) {
+                    // updateState(CHANNEL_SHOWNOTIFICATION, new StringType(""));
+                }
+                break;
+            case CHANNEL_INPUT:
+                if (command instanceof StringType) {
+                    connection.input(command.toString());
+                    updateState(CHANNEL_INPUT, UnDefType.UNDEF);
+                } else if (command.equals(RefreshType.REFRESH)) {
+                    updateState(CHANNEL_INPUT, UnDefType.UNDEF);
+                }
+                break;
+            case CHANNEL_INPUTTEXT:
+                if (command instanceof StringType) {
+                    connection.inputText(command.toString());
+                    updateState(CHANNEL_INPUTTEXT, UnDefType.UNDEF);
+                } else if (command.equals(RefreshType.REFRESH)) {
+                    updateState(CHANNEL_INPUTTEXT, UnDefType.UNDEF);
+                }
+                break;
+            case CHANNEL_ARTIST:
+                if (command.equals(RefreshType.REFRESH)) {
+                    connection.updatePlayerStatus();
+                }
+                break;
+            case CHANNEL_ALBUM:
+                if (command.equals(RefreshType.REFRESH)) {
+                    connection.updatePlayerStatus();
+                }
+                break;
+            case CHANNEL_TITLE:
+                if (command.equals(RefreshType.REFRESH)) {
+                    connection.updatePlayerStatus();
+                }
+                break;
+            case CHANNEL_MEDIATYPE:
+                if (command.equals(RefreshType.REFRESH)) {
+                    connection.updatePlayerStatus();
+                }
+                break;
+            default:
+                logger.debug("Received unknown channel {}", channelUID.getIdWithoutGroup());
+                break;
+        }
+
+    }
+
+    public void playURI(Command command) {
+        connection.playURI(command.toString());
+    }
+
+    public void playNotificationSoundURI(Command command) {
+        connection.playNotificationSoundURI(command.toString());
+    }
+
+    public PercentType getNotificationSoundVolume() {
+        return new PercentType(connection.getVolume());
+    }
+
+    public void setNotificationSoundVolume(PercentType volume) {
+        connection.setVolume(volume.intValue());
+    }
+
+    @Override
+    public void initialize() {
+
+        try {
+
+            connection.connect(this.getConfig().get(HOST_PARAMETER).toString(),
```

Comment:
you might want to do a sanity check of the parameters here first and set the Thing to OFFLINE CONFIGURATION_ERROR, if something is not right.